frontends/p4/structInitializers.cpp

```diff
@@ -17,43 +17,247 @@ limitations under the License.
 #include "structInitializers.h"
 
 namespace P4 {
+const IR::Expression*
+convert(const IR::Expression* expression, const IR::Type* type);
+
+const IR::Expression* defaultValue(const IR::Type* type, const Util::SourceInfo srcInfo) {
+    IR::Expression* expr = nullptr;
+    if (type->is<IR::Type_Bits>()) {
+        expr = new IR::Constant(type, 0);
+    } else if (type->is<IR::Type_Boolean>()) {
+        expr = new IR::BoolLiteral(false);
+    } else if (type->is<IR::Type_Error>()) {
+        auto path = new IR::Path(IR::ID(srcInfo, "error"));
+        auto typeName =  new IR::Type_Name(srcInfo, path);
+        auto id = new IR::ID(srcInfo, "NoError");
+        auto tne = new IR::TypeNameExpression(srcInfo, typeName);
+        expr = new IR::Member(srcInfo, tne, *id);
+    } else if (auto t = type->to<IR::Type_Enum>()) {
+        BUG_CHECK(t->members.size(), "unknown default value for enum with no members");
```

Comment:
I'm not sure how can the user declare an enum with no members.
Wouldn't parser complain because it has an enum declaration defined as a list of names that cannot be empty?

Reply:
I'm not sure how the user can declare an enum with no members.
Wouldn't parser complain because it has an enum declaration with a list of names that cannot be empty?

Reply:
If the spec says that it can't be empty then having a BUG is ok.